src/python/foglamp/core/core_daemon.py

```diff
@@ -179,44 +160,64 @@ def get_pid():
     return pid
 
 
-def safe_makedirs(directory):
+def _safe_makedirs(directory):
     """
     :param directory: working directory
     """
 
-    directory = os.path.expanduser(directory)
     try:
         os.makedirs(directory, 0o750)
     except Exception as e:
         if not os.path.exists(directory):
             raise e
 
 
-def main():
-    safe_makedirs(WORKING_DIR)
-    safe_makedirs(os.path.dirname(PIDFILE))
-    safe_makedirs(os.path.dirname(LOGFILE))
+def _do_main():
+    _safe_makedirs(wdir)
+    _safe_makedirs(os.path.dirname(pidf))
+    _safe_makedirs(os.path.dirname(logf))
 
     if len(sys.argv) == 1:
-        start()
+        print("Usage: start|stop|restart|status|info")
+        # TODO throw an exception instead
+        sys.exit(2)
     elif len(sys.argv) == 2:
-            if 'start' == sys.argv[1]:
-                start()
-            elif 'stop' == sys.argv[1]:
-                stop()
-            elif 'restart' == sys.argv[1]:
-                restart()
-            elif 'status' == sys.argv[1]:
-                print(is_running())
-            elif 'info' == sys.argv[1]:
-                print(get_pid())
+        if 'start' == sys.argv[1]:
```

Comment:
out of this PR's scope, but just for note.. we need to check `sys.argv` vs `argparser`